config/core/resources/channel.yaml

```diff
@@ -51,206 +49,147 @@ spec:
+                    'my-project-1234' rather than its display name, 'My Project' or its number
+                    '1234567890'. If omitted uses the Project ID from the GKE cluster metadata service.
+                subscribers:
+                  type: array
+                  items:
+                    type: object
+                    required:
+                      - uid
+                    properties:
+                      uid:
+                        type: string
+                        minLength: 1
+                      generation:
+                        type: integer
+                      subscriberUri:
+                        type: string
+                      replyUri:
+                        type: string
+                      delivery:
+                        type: object
+                        properties:
+                          deadLetterSink:
+                            type: object
+                            properties:
+                              ref:
+                                type: object
+                                properties:
+                                  kind:
+                                    type: string
+                                  namespace:
+                                    type: string
+                                  name:
+                                    type: string
+                                  apiVersion:
+                                    type: string
+                              uri:
+                                type: string
+                          retry:
+                            type: integer
+                          backoffPolicy:
+                            type: string
+                          backoffDelay:
+                            type: string
+            status:
+              type: object
+              properties:
+                # No subscribable in v1beta1
+                observedGeneration:
+                  type: integer
+                  format: int64
+                conditions:
+                  type: array
+                  items:
+                    type: object
+                    properties:
+                      lastTransitionTime:
+                        # We use a string in the stored object but a wrapper object at runtime.
+                        type: string
+                      message:
+                        type: string
+                      reason:
+                        type: string
+                      severity:
+                        type: string
+                      status:
+                        type: string
+                      type:
+                        type: string
+                    required:
+                      - type
+                      - status
+                address:
+                  type: object
                   properties:
-                    # No deadLetterSink in subscribers properties in v1beta1
```

Comment:
Looks like in the v1beta1, you added deadLetterSink.

Reply:
The comment is a bit misleading, actually deadLetterSink exists in v1beta1: https://github.com/google/knative-gcp/blob/master/vendor/knative.dev/eventing/pkg/apis/duck/v1beta1/delivery_types.go#L28-L34 

This is a bit of a tricky CRD to modify, I'm wondering if we could have a more automated way to do this kind of deprecation, the best I can do is to use [yq](https://github.com/mikefarah/yq) and [jq](https://stedolan.github.io/jq/), the idea is to expand the [merge key](https://yaml.org/type/merge.html) in v1beta1 schema, output it in json, sort the keys and compare the output:

We use the [path expressions](https://mikefarah.gitbook.io/yq/usage/path-expressions) in yq to extract just v1beta1 part in the yaml. 
```bash
yq r -X config/core/resources/channel.yaml 'spec.versions.(name==v1beta1)'
```

Next we need to output it to json, and sort the json by key to eliminate any difference that are caused by the key order. 

```bash
yq r -X config/core/resources/channel.yaml 'spec.versions.(name==v1beta1)' -jP | jq -S
```

Lastly, we can download the old version of `channel.yaml` , do the same thing and compare them:

```bash
diff <(yq r -X old-channel.yaml 'spec.versions.(name==v1beta1)' -jP | jq -S) \
     <(yq r -X config/core/resources/channel.yaml 'spec.versions.(name==v1beta1)' -jP | jq -S)

# there's no output from my side, meaning no diff
# so the v1beta1 in this PR is equivalent to what it was before. 
```

Reply:
Maybe we should consider this https://github.com/google/knative-gcp/issues/920